imap/imap.c

```diff
@@ -59,9 +59,7 @@
 #include "pattern.h"
 #include "progress.h"
 #include "sort.h"
-#ifdef USE_HCACHE
```

Comment:
With the removal of a bunch of `#ifdef USE_HCACHE`, won't we encounter issues if someone compiles without `bdb`,  `gdbm`, `lmdb`, `kyotocabinet`, and `qdbm`?

Reply:
It shouldn't be a problem.  `hcache/hcache.h` only contains a definitions and has no dependencies (no database stuff).

We only need the typedef for `header_cache_t` if the header cache is disabled.